### DISCUSSION TITLE
Matching state in invoked machine from parent machine

### DISCUSSION THREAD

**Discussion by tjdam** [statelyai/xstate#2376] [Q&A]
Hi,

I've been rewriting some of my machines, breaking them into smaller machines which has been really good for code readability, but I've stumbling into some issues trying to access information from the parent machine.

Is there any way I could `state.match` (or something of that effect) from the utmost parent `state` object and get the current state within machines it invokes?

I feel like this is a question that someone would have asked before. Maybe I was limited by my vocabulary in trying to find an answer, I apologize in advance if it's a duplicate.

Thank you!

**Comment by davidkpiano** [comment]
Are you trying to read the **parent state** from a **child machine**, or a **child state** from a **parent machine**?

Regardless, the best practice is to send the relevant state in an event.

To send parent state to child, you can use the [`send(...)` action creator](https://xstate.js.org/docs/guides/actors.html#sending-events-to-actors):

```ts
entry: send(ctx => ({
  type: 'SOME_EVENT',
  data: ctx.someData
}), { to: 'some-actor' })
```

To send child state to parent, use `sendParent(...)`:

```ts
entry: sendParent(ctx => ({
  type: 'SOME_EVENT',
  data: ctx.someData
}))
```

To read last emitted child value (which represents its state), use `.getSnapshot`:

```ts
entry: (ctx) => {
  ctx.someActorRef.getSnapshot(); // some value
}
```

**Reply by tjdam** [reply]
I'm trying to read child state from parent machine!

So you are saying is that I should keep some context data that updates triggered by `sendParent` send by the children if I want to track the state in the children from the parent, yeah?

I was under the impression that I would be able to access children states from parents like they were part of the same machine, like `state.matches('parentState.childState.grandChildState')`.

Are things like this by design?
